graphql-test/src/main/java/org/jahia/test/graphql/GraphQLPublicationTest.java

```diff
@@ -166,28 +169,28 @@ private void testPublication(boolean publishSubNodes) throws RepositoryException
                 + "}"
         );
 
-        JSONObject mutationResult = result.getJSONObject("data").getJSONObject("jcr").getJSONObject("mutateNode");
-        Assert.assertTrue(mutationResult.getBoolean("publish"));
+        SchedulerService schedulerService = BundleUtils.getOsgiService(SchedulerService.class, null);
 
-        // Wait until the node is published via a background job.
         long startedWaitingAt = System.currentTimeMillis();
-        do {
+
+        while(schedulerService.getAllActiveJobs().stream().anyMatch(job -> job.getDescription().equals("Publication"))){
             if (System.currentTimeMillis() - startedWaitingAt > TIMEOUT_WAITING_FOR_PUBLICATION) {
                 Assert.fail("Timeout waiting for node to be published");
             }
+        }
+
+        JSONObject mutationResult = result.getJSONObject("data").getJSONObject("jcr").getJSONObject("mutateNode");
+        Assert.assertTrue(mutationResult.getBoolean("publish"));
+
+        // Wait until the node is published via a background job.
+        do {
```

Comment:
You don't  need the do/while statement here as you already know that no publication job is running.